posthog/session_recordings/session_recording_helpers.py

```diff
@@ -103,12 +103,28 @@ def split_replay_events(events: list[Event]) -> tuple[list[Event], list[Event]]:
 
 
 # TODO is this covered by enough tests post-blob ingester rollout
-def preprocess_replay_events_for_blob_ingestion(events: list[Event], max_size_bytes=1024 * 1024) -> list[Event]:
-    return _process_windowed_events(events, lambda x: preprocess_replay_events(x, max_size_bytes=max_size_bytes))
+def preprocess_replay_events_for_blob_ingestion(
+    events: list[Event], max_size_bytes=1024 * 1024, user_agent: str | None = None
+) -> list[Event]:
+    return _process_windowed_events(
+        events, lambda x: preprocess_replay_events(x, max_size_bytes=max_size_bytes, user_agent=user_agent)
+    )
+
+
+def snapshot_library_fallback_from(user_agent: str | None) -> str | None:
+    if user_agent is None:
+        fallback = None
+    else:
+        fallback = "web"
+        if "posthog" in user_agent:
+            # mobile SDKs send e.g. posthog-android
+            fallback = user_agent
```

Comment:
```suggestion
            fallback = user_agent.split("/")[0]
```

well have to handle the case that 0 does not exist etc

Reply:
yep, wrapped it in a try and did some fangling